src/sass/parse.js

```diff
@@ -1786,13 +1785,23 @@ function checkIdent(i) {
     i += l;
   }
 
-  if (!wasIdent && !wasInt && tokens[start].type !== TokenType.Asterisk) return 0;
+  // Check if `-` token should be considered an operator and not an ident
+  if (!wasIdent && !checkIdentHyphenMinus(i, start)) return 0;
+
+  if (!wasIdent &&
+      tokens[start].type !== TokenType.Asterisk &&
+      tokens[start].type !== TokenType.HyphenMinus) return 0;
```

Comment:
That will parse `.-, a {}` as valid code, where `.-` will be a class named `-`.
The issue is somewhat related to #114 and I can't think of a fast and correct solution to propose.

As a hackish fix you can modify `checkClass` and `checkShash` functions to check not only interpolations/idents but also hyphens before/between them. This will fix this specific case and not break anything else.
What do you think?